internal/gatewayapi/route.go

```diff
@@ -1082,6 +1082,14 @@ func (t *Translator) processTCPRouteParentRefs(tcpRoute *TCPRouteContext, resour
 			accepted = true
 			irKey := t.getIRKey(listener.gateway)
 
+			tls := ir.TLS{
```

Comment:
shouldnt this be wrapped inside an `if` 
e.g. https://github.com/envoyproxy/gateway/blob/33443f8846f105adaa57699fdd90202b5670c3ce/internal/gatewayapi/listener.go#L53

Reply:
I don't think anything's changed here, previously we set unconditionally:

```
TLS: &ir.TLS{Terminate: irTLSConfigs(listener.tlsSecrets)},
```

`irTLSConfigs` returns `nil` if there aren't any secrets. The line you linked to is also gated by checking the listener protocol.

Reply:
Cool thanks for the checking!